src/common/misc.c

```diff
@@ -536,6 +536,11 @@ daos_hhash_link_insert(struct d_hlink *hlink, int type)
 {
 	D_ASSERT(daos_ht.dht_hhash != NULL);
 
+	/* check if custom type fits in allocated bits */
+	D_ASSERTF(type < (1 << D_HTYPE_BITS),
```

Comment:
Does this assert happen twice (once in cart and once here) ?  I think it would be better to leave it just to cart.

Reply:
makes sense, I will update this with the new CaRT commit hash once it gets landed